assets/src/edit-story/app/story/useStoryReducer/reducers/combineElements.js

```diff
@@ -92,6 +93,13 @@ function combineElements(state, { firstElement, secondId }) {
   // If the element we're dropping into is not background, maintain link, too.
   if (!secondElement.isBackground) {
     propsFromFirst.push('link');
+    // If the second element is rectangular, maintain border, too.
+    if (
+      !secondElement.mask ||
+      secondElement.mask.type === MaskTypes.RECTANGLE
+    ) {
```

Comment:
Added both -- since we need to actually get the mask object before checking for border support then we couldn't just to `mask.supportsBorder` on the element directly. So added `supportsBorder` param to `RECTANGLE` plus a helper.